static/src/javascripts-legacy/projects/commercial/modules/article-aside-adverts.js

```diff
@@ -15,58 +13,57 @@ define([
     config,
     mediator,
     fastdom,
-    addSlot,
-    createSlot,
     commercialFeatures
 ) {
     var minArticleHeight = 1300;
     var minImmersiveArticleHeight = 600;
 
+    var mainColumnSelector = '.js-content-main-column';
+    var rhColumnSelector = '.js-secondary-column';
+    var adSlotSelector = '.js-ad-slot';
+
+    function minContentHeight() {
+      return config.page.isImmersive ? minImmersiveArticleHeight : minArticleHeight;
+    }
+
     function init(start, stop) {
         start();
 
-        var $col = $('.js-secondary-column');
-        var $mainCol;
+        var $col = $(rhColumnSelector);
+        var $mainCol, $adSlot;
 
         // are article aside ads disabled, or secondary column hidden?
         if (!(commercialFeatures.articleAsideAdverts && $col.length && $css($col, 'display') !== 'none')) {
             stop();
             return Promise.resolve(false);
         }
 
-        $mainCol = $('.js-content-main-column');
+        $mainCol = $(mainColumnSelector);
+        $adSlot = $(adSlotSelector, $col);
+
+        if (!$adSlot.length) {
+            stop();
+            return Promise.resolve(false);
+        }
 
-        fastdom.read(function () {
+        return fastdom.read(function () {
             return $mainCol.dim().height;
         })
         .then(function (mainColHeight) {
-            var adSlot, adType;
-            var $adSlotContainer = $('.js-ad-slot-container', $col[0]);
 
-            if (config.page.isImmersive) {
-                adType = mainColHeight >= minImmersiveArticleHeight ?
-                        'right' :
-                        'right-small';
-            } else {
-                adType = mainColHeight >= minArticleHeight ?
-                    'right-sticky' :
-                    'right-small';
+            // Should switch to 'right-small' MPU for short articles
+            if (mainColHeight < minContentHeight()) {
+              fastdom.write(function () {
+                  $adSlot.removeClass('right-sticky js-sticky-mpu is-sticky');
+                  $adSlot[0].setAttribute('data-mobile', '1,1|2,2|300,250|fluid');
```

Comment:
as much as I don't like bonzo, here you can just write `$adSot.data('mobile', '...')`